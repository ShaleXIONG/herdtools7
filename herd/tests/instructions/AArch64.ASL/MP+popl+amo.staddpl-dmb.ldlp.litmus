AArch64 MP+popl+amo.staddpl-dmb.ldlp
Orig=PodWWPL RfeLP Amo.StAddPL DMB.LDdWRLP Fre
{
0:X1=x; 0:X3=y;
1:X3=y; 1:X1=x;
}
 P0           | P1             ;
 MOV W0,#1    | MOV W0,#2      ;
 STR W0,[X1]  | STADDL W0,[X3] ;
 MOV W2,#1    | DMB LD         ;
 STLR W2,[X3] | LDR W2,[X1]    ;
exists ([y]=3 /\ 1:X2=0)

