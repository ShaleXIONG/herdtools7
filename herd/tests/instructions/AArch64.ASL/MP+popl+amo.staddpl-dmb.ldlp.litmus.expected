Test MP+popl+amo.staddpl-dmb.ldlp Allowed
States 4
1:X2=0; [y]=1;
1:X2=0; [y]=3;
1:X2=1; [y]=1;
1:X2=1; [y]=3;
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists ([y]=3 /\ 1:X2=0)
Observation MP+popl+amo.staddpl-dmb.ldlp Sometimes 1 3
Hash=dcfb25d5ffad6133ce9a386145811bf6

